control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go

```diff
@@ -854,7 +854,7 @@ haproxy -f /tmp/haproxy.conf
 		deploymentConfig.Scheduling.PriorityClass = hcp.Annotations[hyperv1.ControlPlanePriorityClass]
 	}
 	deploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
-	deploymentConfig.SetRequestServingDefaults(hcp, nil, nil)
+	deploymentConfig.SetRequestServingDefaults(hcp, map[string]string{"app": "ignition-server-proxy"}, nil)
```

Comment:
any chance we could create a variable for the label rather than hardcoding it in here?

Reply:
Good questions, a variable would cause the underlying functions manipulating references to the map accidentally changing every field pointing to the variable. We could thought have a function returning a new map on each invocation instead - we do that already in some components. FWIW This whole library support.config library need refactor soonish to be properly used by componentes as a gating library enforcing hypershift opinions. We'll want to do that refactor after we have e2e coverage for all those opinions that you're looking into.